framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -339,106 +339,125 @@ public TransferFunction createFlowTransferFunction(CFAbstractAnalysis<Value, Sto
     }
 
     /**
-     * Create {@link QualifierDefaults} which handles user specified defaults
-     * @return the QualifierDefaults class
+     * Create {@link QualifierDefaults} which handles checker specified defaults
+     * Subclasses should override {@code addCheckedCodeDefaults(QualifierDefaults defs)} or
+     * {@code addUncheckedCodeDefaults(QualifierDefaults defs)} to add more defaults or
+     * use different defaults.
      *
-     * TODO: should this be split in two methods to allow separate reuse?
+     * @return the QualifierDefaults class
      */
-    protected QualifierDefaults createQualifierDefaults() {
+    protected final QualifierDefaults createQualifierDefaults() {
         QualifierDefaults defs = new QualifierDefaults(elements, this);
-        boolean foundDefaultOtherwise = false;
-        boolean foundDefaultOtherwiseForUnannotatedCode = false;
+        addCheckedCodeDefaults(defs);
+        addUncheckedCodeDefaults(defs);
+        return defs;
+    }
 
-        // TODO: Verify that only one default per location type is present.
+    /**
+     * Adds default qualifiers for type-checked code by
+     * reading  {@code @DefaultFor} and {@code @DefaultQualifierInHierarchy}
+     * meta-annotations.
+     * Subclasses may override this method to add defaults that cannot be specified with
+     * a {@code @DefaultFor} or {@code @DefaultQualifierInHierarchy} meta-annotations.
+     *
+     * @param defs QualifierDefault object to which defaults are added
+     */
+    protected void addCheckedCodeDefaults(QualifierDefaults defs) {
+        boolean foundOtherwise = false;
+        // Add defaults from @DefaultFor and @DefaultQualiferInHierarchy
         for (Class<? extends Annotation> qual : getSupportedTypeQualifiers()) {
             DefaultFor defaultFor = qual.getAnnotation(DefaultFor.class);
             if (defaultFor != null) {
-                final DefaultLocation [] locations = defaultFor.value();
-                defs.addCheckedCodeDefaults(AnnotationUtils.fromClass(elements, qual), locations);
-
-                foundDefaultOtherwise = foundDefaultOtherwise ||
-                        Arrays.asList(locations).contains(DefaultLocation.OTHERWISE);
+                final DefaultLocation[] locations = defaultFor.value();
+                defs.addCheckedCodeDefaults(AnnotationUtils
```

Comment:
I would undo the added linebreaks on this line. (I would rather keep the line-break on the next line - which is much longer than this line...)